sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogManager.scala

```diff
@@ -46,6 +47,7 @@ class CatalogManager(
   private val catalogs = mutable.HashMap.empty[String, CatalogPlugin]
 
   def catalog(name: String): CatalogPlugin = synchronized {
+    validateName(name)
```

Comment:
Instead of reusing `invalidDatabaseObjectNameError` (not sure if catalog should be considered as database object), should we have another exception such as `InvalidCatalogNameException` and throw it in `Catalogs.load`? we can also give more informative error message such as: "The catalog entry is defined and mapped to XXX but the catalog name is of invalid format" etc.

Reply:
I think if we want keep consistent with the namespace/table naming rule, We have already considered `catalog` as a database object. And I agree with you that we should have another exception `InvalidCatalogNameException`, because the error message is more clarify.